launcher/ui/pages/instance/ScreenshotsPage.cpp

```diff
@@ -174,7 +170,7 @@ class FilterModel : public QIdentityProxyModel
             }
             if (!m_failed.contains(filePath))
             {
-                ((FilterModel *)this)->thumbnailImage(filePath);
+                const_cast<FilterModel*>(this)->thumbnailImage(filePath);
```

Comment:
> `// this is about as elegant and well written as a bag of bricks with scribbles done by insane`
> `// asylum patients.`
> Jan Dalheimer, 8 years ago

Got to love tech dept.

Anyway, you could get rid of the `const_cast`s (including the ones a couple lines above) by making `thumbnailImage` const and `m_thumbnailCache`, `m_thumbnailingPool`, `watched`, and `watcher` mutable, but it'd probably be better to take a longer look at this class and redesign now or at a later time.